### PR TITLE
Wallet features / refactoring

### PR DESCRIPTION
**Pull request by lock9**
Wif Decryption Documentation
Account refactoring
Unit tests
EncryptWif
Password for wallet operations;
Replacing String with SecureString when handling passwords;
TransactionManager (initial)
Genesis Block Refactoring (review required)
Refactoring code following Aboim feedback;
Replacing references to ICrypto;
Wrappers for FileInfo and JSONConverter

### CODE REVIEW

File: src/NeoSharp.Core/Wallet/NEP6/NEP6Account.cs [hunk 1]
```diff
@@ -18,5 +18,11 @@ public class NEP6Account : IWalletAccount
 
         [JsonProperty("extra")]
         public Object Extra { get; set; }
+
+        //TODO: Refactor
+        public object Clone()
+        {
+            throw new NotImplementedException();
```
Comment (aboimpinto) [inline src/NeoSharp.Core/Wallet/NEP6/NEP6Account.cs]: when this is going to be implemented?

File: src/NeoSharp.Core/Wallet/Helpers/Extensions.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                isValid = false;
+            }
+
+            if(privateKey[33] != 0x01){
+                isValid = false;
+            }
+
+            return isValid;
+        }
+
```
Comment (lock9) [inline src/NeoSharp.Core/Wallet/Helpers/Extensions.cs:48]: How to create validation methods that retain error messages? If I return bool, the string is lost. If I throw exceptions, the methods looks weird (never returns false).
Suggestions?

File: src/NeoSharp.Core/Wallet/NEP6/NEP6WalletManager.cs [hunk 3]
```diff
@@ -407,19 +409,13 @@ private IWalletAccount TryGetAccount(UInt160 accountScriptHash)
         }
 
         /// <summary>
-        /// Load a wallet using a fileInfo.
+        /// Load a wallet using a file name.
         /// </summary>
-        /// <param name="fileInfo">File info.</param>
-        public void Load(FileInfo fileInfo)
+        /// <param name="fileName">File name.</param>
+        public void Load(String fileName)
```
Comment (aboimpinto) [inline src/NeoSharp.Core/Wallet/NEP6/NEP6WalletManager.cs]: always use "string" instead "String"
check the comments here

https://stackoverflow.com/questions/7074/what-is-the-difference-between-string-and-string-in-c

**Review by aboimpinto — APPROVED** [review]
there are a lot of changes .. would be nice for the next PR have several PRs

File: src/NeoSharp.Core/Wallet/Helpers/Extensions.cs (hunk 2 above)
Comment (osmirnov) [reply src/NeoSharp.Core/Wallet/Helpers/Extensions.cs:48]: There is an approach to have a pair of validation methods:
1. bool Validate(payload) // return true or throws an error
2. bool TryValidate(payload, out error) // return true/false and error as the second argument

**Comment by osmirnov** [comment]
Please resolve the conflicts by merging CityOfZion:development into simplitech:wallet2.

**Pull request merged**